diff --git a/catboost/python-package/setup.py b/catboost/python-package/setup.py
index fe9251a21f..7e129aca23 100644
--- a/catboost/python-package/setup.py
+++ b/catboost/python-package/setup.py
@@ -222,11 +222,11 @@ class build_ext(_build_ext):

     def build_with_make(self, topsrc_dir, build_dir, catboost_ext, put_dir, verbose, dry_run):
         logging.info('Buildling {} with gnu make'.format(catboost_ext))
-        makefile = 'python{}.{}CLANG11-LINUX-X86_64.makefile'.format(python_version()[0], 'CUDA.' if self.with_cuda else '')
+        makefile = 'python{}.{}CLANG12-LINUX-X86_64.makefile'.format(python_version()[0], 'CUDA.' if self.with_cuda else '')
         make_cmd = [
             'make', '-f', '../../make/' + makefile,
-            'CC=clang-11',
-            'CXX=clang++-11',
+            'CC=clang',
+            'CXX=clang++',
             'BUILD_ROOT=' + build_dir,
             'SOURCE_ROOT=' + topsrc_dir,
         ]
